### ISSUE TITLE
Implement Admin Trends API

### ISSUE THREAD

**Issue by andregasser** [andregasser/bigbone#326]
Mastodon's Admin Trends API allows an admin to view trending links, statuses and tags. The goal of this task is to implement the following endpoints of this API in BigBone:

- View trending links
- View trending statuses
- View trending tags

The Admin Reports API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/trends/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

**Comment by andregasser** [comment]
Please do not start implementing this issue before https://github.com/andregasser/bigbone/issues/297 is merged into master, as some base entities will be created in the other story that are needed here as well (e.g.. Trend::Link).
